### DISCUSSION TITLE
Include GNSS support instead of only GPS

### DISCUSSION THREAD

**Discussion by Corsair-63** [peterantypas/maiana#26] [Ideas]
first of all I just discovered your project and it sound really marvellous.

after that of course I have some ideas, but no clue of programming, but I really can help as tester, because I'm well familiarized with NMEA comms on board.

why not include in the positioning chip the capability for GNSS?? 
I mean at least include GLONASS, Galileo, GPS, Beidou, ....

I guess it should be really fine to make it positioning redundant.

I'll try to share some other ideas for your consideration.

Cheeer

**Comment by peterantypas** [comment]
I did have a GNSS originally. It was the Telit SE873 and it was awesome, until they decided to replace it with the SE873Q5, which contains a switching regulator that literally craps all over the AIS frequencies and degrades RX sensitivity to unacceptable levels.

I'm open to exploring GNSS units eventually, but right now, the Quectel GPS with the MediaTek 3333 chip is outstanding and dirt cheap.

**Reply by Corsair-63** [reply]
Thanks for the reply and the course of new chips

**Reply by peterantypas** [reply]
There may be more options available now that I was forced to adopt the Quectel module footprint, as there are several MediaTek based GPS/GNSS modules that could be drop-ins. I looked at a few and they seemed to have a compatible pinout.

**Comment by peterantypas** [comment]
After spending some time doing A/B tests between the Telit and Quectel, I have to admit that the latter doesn't compare. The MTK3333 can't stand still: It seems to always move randomly by a couple of meters at least, and its SOG is never zero. By contrast, once the Telit acquires enough satellites it stands still, with zero SOG.

I am going to try a radical do-over of the layout to see if I can get the Telit back in the design without sacrificing AIS reception too much. The Telit has a plastic cap; it may need a dedicated shield can over it.

**Comment by peterantypas** [comment]
I'm also going to try a couple of GNSS modules that fit the footprint and should be drop-ins: Quectel L76 and Allystar TAU1105. The last one is interesting, because it's not MediaTek based, it's a completely different chipset.

**Comment by junkt1** [comment]
Have you ever tried ublox GNSS modules. 

https://www.u-blox.com/en/positioning-chips-and-modules

I made best experience with such modules in cooperation with BTLE, NBIOT and WiFi including rx sensitivity degradation etc. I live close to ublox head office, have an nda and access to internal details of the midules. If you are interested or need any help let me know. 

PS I appreciate a lot what you do here. I’m planning to add POE to your system when it’s finished and stable.

Regards 

Thom

Lat./Long.: N 47 12'45.9" / E 007 32'41.1"

always look on the bright side of life 😊

**Reply by peterantypas** [reply]
I would definitely try ublox if they didn't cost 2x as much as Telit and 4x as much as all the Chinese ones. Since you already have an NDA, can you find out if they contain DC/DC switches that raise the noise floor at 162MHz?

**Comment by peterantypas** [comment]
A bit of good news: I constructed a shield can over the SE873Q5 using kapton and copper tape, and it did make a difference. So it looks like at least that option may be viable.

**Comment by peterantypas** [comment]
I just tested the Quectel L76 as a direct drop-in replacement and it seems to work pretty well. I counted 12 GPS and 7 GLONASS satellites at some point. After about 1 hour of continuous operation, it "froze" in place with a SOG of less than 0.1kts, so at least it got that right.

As far as I'm concerned, design of the main transponder board is complete with rev 10.0.1 as posted in the repository. Any Quectel L7x series module with a standard 9.7x10.1 form factor will work.

**Comment by peterantypas** [comment]
A more comprehensive test with board 9.3 proved that the Quectel L76 is in fact just as good as the Telit SE873 without the EMC side effects. This is what I will bundle into the final kit.

**Reply by rszemeti** [reply]
Hi Peter,   Just pulling in the final parts before I paste the board ... is this now the preferred device:

https://uk.rs-online.com/web/p/gnss-gps-modules/9084108/

**Comment by peterantypas** [comment]
Yes, this is it

**Comment by peterantypas** [comment]
I just tested the Quectel L76**L**-M33 module (which adds an extra input LNA), and noticed a significant improvement in horizontal accuracy. It's now routinely below 1 meter, and SOG seems to be quite accurate as well (especially with a mild low pass filter).

Next week I will also test the Ublox MAX-M8Q-0 and report back. Adding this module will require a board spin though, because it has a "VIO" input that needs to touch the 3.3V rail. It's also possible that it might desense the AIS receiver because it contains a DC/DC switch (albeit under a metal shield can).

**Reply by revolteur** [reply]
That's Great! Would be nice to hear how the Ublox SOG ends up performing. Having accurate SOG would be nice when the good old knot meter stops working.

**Comment by peterantypas** [comment]
While building evaluation boards for different GNSS modules, I realized that something seemed off: One of those modules would track 7-8 GLONASS satellites and 0 GPS satellites. 

While debugging this, I decided to revisit the GPS antenna tuning, and then it dawned on me: When I originally designed the matching network for it, the board was laying flat on a wooden table, but it wasn't mounted inside the PVC enclosure. So I drilled through a scrap enclosure to pass a pigtail cable and then proceeded to tune the antenna again. Sure enough, the enclosure changes the tuning substantially, to the point where it's rather poor at the L1 band.

Going forward, all new boards will be hand-modified with a different matching network for the GPS antenna.

**Reply by peterantypas** [reply]
Here is the new tuning, measured inside the enclosure, with the VHF antenna attached:

![Screenshot from 2021-09-14 20-56-58](https://user-images.githubusercontent.com/1565933/133486372-cd7510f0-1cbf-4e3e-977b-0ba3cb8391aa.png)

**Reply by peterantypas** [reply]
To take things further, I need to construct a slightly more elaborate tuning setup which allows me to measure the antenna impedance with the board wrapped in clear heatshrink and encased in the PVC enclosure, because that's exactly how the kit is assembled.

**Comment by rszemeti** [comment]
Those Nano VNA boards are excellent value really. The equivalent HP
instrument would cost as much as a good car!

I am however a sucker for nice test gear, this weeks purchase was an
HP8970B and an Eaton 2075 ... I managed to resist a very nice HP 8563A
analyser, but I might give in to the temptation!

**Comment by rszemeti** [comment]
Have you tried SimSmith? I think the latest version will take data directly
out of NanoVNA ... I've found it useful for solving various matching
problems.  I guess in the end it comes down to fiddling with component
values and a solderign iron, but SimSmith should get you pretty close.

http://www.ae6ty.com/Smith_Charts.html

**Reply by peterantypas** [reply]
I use an online Smith chart tool here: https://www.rfmentor.com/jSmith but I could definitely give SimSmith a try.

But the main challenge for me is to set up the measurement to be as unobtrusive as possible, then fiddle with component values. In the end I just look at GSV sentences and average C/No values between GPS and GLONASS and that gives me a pretty good idea of where it stands.

**Comment by rszemeti** [comment]
I've had a lot of success using it to design amplifier secions, with .s1p
files for the active device and producing the input and output  matching
circuits ... you can see in real time the input match chnage as you fiddle
with the output ... very easy to use.

**Comment by peterantypas** [comment]
Alright, this is done. The main lesson learned here is that the PCBA should not be covered by any heatshrink material because it severely de-tunes the antenna and no matching network will compensate for it.

With the help of some spare boards, I was able to calibrate the VNA with the pigtail cable attached, then measured inside the enclosure by drilling a hole

![IMG_1293](https://user-images.githubusercontent.com/1565933/133855051-0f13e910-c392-4e89-b5fd-9e4c0fc9bd5b.JPG)

The rest was some combination of Smith charting and trial/error. But I got it:

![2 5pF-5 6nH-4pF-Module-NoHeatShrink](https://user-images.githubusercontent.com/1565933/133855243-bb30db64-f6ad-4e11-84d2-05909b10702f.png)

**Reply by peterantypas** [reply]
Now I'm running the unit inside its enclosure. It's indoors under my "solarium" so it has pretty good sky exposure, but nothing like on the boat. Still, it's tracking 11 GPS and 9 GLONASS satellites which is reasonable for the Quectel L76.

**Comment by peterantypas** [comment]
I was also able to (briefly) play with the Quectel LC79D. This device is very different from the others. It's based on a Broadcom chipset and it was able to track GPS, Galileo and GLONASS vehicles during my brief test. Cost is 3x of L76.

I have yet to assemble a board with the Ublox, that's next. I was only able to purchase the MAX-7C-0, nothing else. Cost is 4x the L76.

**Comment by rszemeti** [comment]
I think access to the other 2 systems will bring better performance than
anyting you can do to the antenna. You already have return loss better than
-10 dB, any improvements from here will be very small.

I managed to avoid buying any more test gear today ... apart from an HP
vector voltmeter for setting up some -3 dB hybrids ...

**Reply by peterantypas** [reply]
I agree. I just assembled the board with the Ublox and I can tell its stationary drift indoors is better than the L76. I'm going to the boat today and I will do a shootout between MAX-7 and L76L. The LC79D is a 1.8V device, needs a modified transponder board with a level shifter so no can do today ...

Do you have enough walking room around your test gear?

**Comment by rszemeti** [comment]
The test gear doesn't take up that much room ... it's the pick and place
machine and conveyor oven that really hog the floor space ;)  The pick and
place hogs bench space too, as loading and unloading feeders is a huge
pain, and I always get into a mess.

**Comment by peterantypas** [comment]
Well, that was quick! This test reaffirmed what I always suspected, but now I have the evidence to support it.

TL;DR. The Ublox C-7 is an overpriced and underwhelming offering.

Test methodology: Two identical boards, one with Quectel L76 and one with Ublox C-7 are mounted into the exact same MAIANA enclosure (complete with VHF antenna) and run through a stationary drift test for about 30 minutes each. The test begins by letting each GNSS run for 5 minutes to acquire a fix and build enough satellite almanac information so as to give each a fair chance. Then, I start tracking their position and saving the track as KML. The goal is to register very little (ideally zero) motion.

This test was conducted with the new antenna tuning. Both boards had the exact same matching network.

The result? The $5 L76 beat the living crap out of the $20 Ublox C-7. Like they're not even in the same league!

Quectel track (I gave up after half an hour because it wouldn't move anymore):

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<kml xmlns:atom="http://www.w3.org/2005/Atom" xmlns="http://www.opengis.net/kml/2.2" xmlns:gx="http://www.google.com/kml/ext/2.2" xmlns:kml="http://www.opengis.net/kml/2.2">
  <Document>
    <name>OpenCPN Track</name>
    <Placemark>
      <name></name>
      <gx:Track>
        <when>2021-09-18T18:24:30Z</when>
        <when>2021-09-18T18:40:32Z</when>
        <gx:coord>-122.317090 37.868078 0.0</gx:coord>
        <gx:coord>-122.317090 37.868063 0.0</gx:coord>
      </gx:Track>
    </Placemark>
  </Document>
</kml>
```

Ublox track:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<kml xmlns:atom="http://www.w3.org/2005/Atom" xmlns="http://www.opengis.net/kml/2.2" xmlns:gx="http://www.google.com/kml/ext/2.2" xmlns:kml="http://www.opengis.net/kml/2.2">
  <Document>
    <name>OpenCPN Track</name>
    <Placemark>
      <name></name>
      <gx:Track>
        <when>2021-09-18T17:28:46Z</when>
        <when>2021-09-18T17:32:18Z</when>
        <when>2021-09-18T17:33:30Z</when>
        <when>2021-09-18T17:37:30Z</when>
        <when>2021-09-18T17:38:28Z</when>
        <when>2021-09-18T17:42:56Z</when>
        <when>2021-09-18T17:44:18Z</when>
        <when>2021-09-18T17:49:55Z</when>
        <when>2021-09-18T17:57:07Z</when>
        <gx:coord>-122.317106 37.868047 0.0</gx:coord>
        <gx:coord>-122.317070 37.868031 0.0</gx:coord>
        <gx:coord>-122.317083 37.868065 0.0</gx:coord>
        <gx:coord>-122.317122 37.868042 0.0</gx:coord>
        <gx:coord>-122.317088 37.868014 0.0</gx:coord>
        <gx:coord>-122.317130 37.868060 0.0</gx:coord>
        <gx:coord>-122.317110 37.868034 0.0</gx:coord>
        <gx:coord>-122.317107 37.868080 0.0</gx:coord>
        <gx:coord>-122.317118 37.868049 0.0</gx:coord>
      </gx:Track>
    </Placemark>
  </Document>
</kml>

```

**Reply by peterantypas** [reply]
So why is this? I think it's because UBLOX devices are not as sensitive. Reading through the Hardware Integration documents for several of them, it becomes clear very quickly that they expect an *active antenna or external LNA*.

The C-7 was not able to acquire a single GLONASS satellite during this test. The Quectel had 7. Could the C-7 match this performance if it had an active antenna? Maybe so, but why bother? Why add yet more BOM and complexity to this board? 

Case closed, as far as I'm concerned.

Now, the performance of the L76 has improved since the last time I run a similar test, due to the new antenna tuning. So I'm going to redesign the MAIANA board with the new GNSS matching network components and call it a day :)


P.S. This test was conducted with the L76-M33, *not* the L76L-M33 (which has an extra LNA at its input). I'll repeat this test with that variant later just for kicks.

**Comment by rszemeti** [comment]
That's impressive. I can see why the uBlox is built that way,  as many
applications do use an external aerial with built-in LNA, and in the past,
there was not a whole lot of strong signals in that band, but these days,
there are, and they probably wanted to limit front-end gain to prevent
blocking when used in an urban environment ... but, for our application,
adding an LNA device with bias and decoupling is probably not a great plan.

Out of interest, what is the "cold start" time on the L76?  Just idly
thinking about PLB type applications ... the board is small enough ..

**Reply by peterantypas** [reply]
Yeah, I think it makes sense for the applications they target. The L76 typically takes < 1 minute cold start outdoors.

**Comment by rszemeti** [comment]
That's a pretty decent cold start time I guess,  it would make a PLB very
feasible.

**Reply by peterantypas** [reply]
The L76L will be even better here. You definitely don't have to worry about input stage gain where you'll be when you deploy a PLB ;)

**Comment by peterantypas** [comment]
I was unable to record a track with the L76L. It did not move. At all. In over 45 minutes. We have a winner!

**Comment by rszemeti** [comment]
That's interesting ... I shall have a look at that device for another
project, a GPS disciplined 10 mhz reference ... it just needs a very good,
low jitter 1pps output to drive the reference comparator.  If the
positioning accuracy is good on this device, there is every chance the 1
pps output is low jitter also.

**Comment by peterantypas** [comment]
They claim 10ns accuracy for 1pps

**Comment by peterantypas** [comment]
Alright, this is it. It's going into the next board design.

![IMG_1301](https://user-images.githubusercontent.com/1565933/134206102-5a9b80da-4b22-49bf-b4f7-46739ba0dedc.JPG)

**Comment by rszemeti** [comment]
As my bits are still sitting in a box ... are you any closer to having some
completed units available for release?

**Reply by peterantypas** [reply]
I've shipped 36 and have another 3 in the oven ;)

DM me at maiana.kits __at__ gmail d0t com
